hyperspy/_signals/eds_tem.py

```diff
@@ -325,12 +363,33 @@ def quantification(self,
             navigation_mask = navigation_mask.data
         xray_lines = self.metadata.Sample.xray_lines
         composition = utils.stack(intensities)
-        composition.data = utils_eds.quantification_cliff_lorimer(
-            composition.data, kfactors=kfactors,
-            mask=navigation_mask) * 100.
-        composition = composition.split()
+        if method == 'CL':
+            composition.data = utils_eds.quantification_cliff_lorimer(
+                composition.data, kfactors=factors,
+                mask=navigation_mask) * 100.
+        elif method == 'zeta':
+            results = utils_eds.quantification_zeta_factor(
+                composition.data, zfactors=factors,
+                dose=self._get_dose(method))
+            composition.data = results[0] * 100.
+            mass_thickness = intensities[0].deepcopy()
+            mass_thickness.data = results[1]
+            mass_thickness.metadata.General.title = 'Mass thickness'
+        else:
```

Comment:
I agree that forcing the user to specify the method is better. One
advantage with HyperSpy is that the user must think and not just "click a
button". :)
11. apr. 2016 15:23 skrev "k8macarthur" <notifications@github.com> følgende:

> In hyperspy/_signals/eds_tem.py
> <https://github.com/hyperspy/hyperspy/pull/913#discussion_r59202332>:
>
> > -            composition.data, kfactors=kfactors,
> > -            mask=navigation_mask) * 100.
> > -        composition = composition.split()
> > +        if method == 'CL':
> > +            composition.data = utils_eds.quantification_cliff_lorimer(
> > +                composition.data, kfactors=factors,
> > +                mask=navigation_mask) * 100.
> > +        elif method == 'zeta':
> > +            results = utils_eds.quantification_zeta_factor(
> > +                composition.data, zfactors=factors,
> > +                dose=self._get_dose(method))
> > +            composition.data = results[0] * 100.
> > +            mass_thickness = intensities[0].deepcopy()
> > +            mass_thickness.data = results[1]
> > +            mass_thickness.metadata.General.title = 'Mass thickness'
> > +        else:
>
> I think I'd rather force them to pick an method so that they have to think
> about which method they've chosen and have the correct standards values for
> that method.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hyperspy/hyperspy/pull/913/files/23ce89b24f0c3f0ca8b5e2cf424c505eb8d6be6a#r59202332>
>